virtualmouse/fakeinput.go

```diff
@@ -1,11 +1,39 @@
 package main
 
 import (
+	"fmt"
 	"github.com/BurntSushi/xgb"
 	"github.com/BurntSushi/xgb/xproto"
 	"github.com/BurntSushi/xgb/xtest"
 )
 
-func fakeinput (X *xgb.Conn, wid xproto.Window, x int16, y int16) {
-	xtest.FakeInput(X, 6, 0, 0, wid, x, y, 0)
+type connInfo struct {
+	 conn *xgb.Conn
+	 wid xproto.Window
+}
+
+func establishConn() connInfo {
```

Comment:
Return (ConnInfo, error) rather than just conninfo (take a look at Leshna's PR to see how errors are used, they are usually returned everywhere in Go.